src/ProtectedProxies/MinimalProtectedProxyFactory.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: UNLICENSED
+// (c) SphereX 2023 Terms&Conditions
+
+pragma solidity ^0.8.0;
+
+import {SphereXConfiguration} from "../SphereXConfiguration.sol";
+import {SpherexProtetedMinimalProxy} from "./ProtectedMinimalProxy.sol";
+
+// TODO: add event for deploying
+
+/**
+ * @title A factory contract that deploys the SpherexProtetedMinimalProxy contracts.
+ */
+contract MinimalProtectedProxyFactory is SphereXConfiguration {
+
+    address private immutable IMPLEMENTATION;
+    address private immutable ALLOWED_DEPLOYER;
+    bytes4[] private /*immutable*/ PROTECTED_SIGS; // this is in effect immutable.
+
+    constructor(address allowedDeployer, address implementation, bytes4[] memory protectedSigs ) SphereXConfiguration(msg.sender, address(0), address(0)) {
```

Comment:
weve created a loop, in order to deploy zeebu factory it needs to know the implementations and the spherex factory, in order to deploy spherex factory it need to know the zeebu factory...we shuold set the alloweddeployer in a different function